### ISSUE TITLE
Integration quit working

### ISSUE THREAD

**Issue by 400HPMustang** [itchannel/fordpass-ha#180]
Hello, today the integration stopped working for both of my vehicles. Here is the information from the logs, my VINs have been redacted.

```
This error originated from a custom integration.

Logger: custom_components.fordpass
Source: helpers/update_coordinator.py:223
Integration: FordPass (documentation, issues)
First occurred: 2:57:00 PM (225 occurrences)
Last logged: 5:13:41 PM

Error fetching fordpass data: Error communicating with FordPass for 3XXXXXXXXXXXXXXX1
Error fetching fordpass data: Error communicating with FordPass for 1XXXXXXXXXXXXXXX7
```

```
This error originated from a custom integration.

Logger: custom_components.fordpass
Source: custom_components/fordpass/__init__.py:189
Integration: FordPass (documentation, issues)
First occurred: 2:57:00 PM (225 occurrences)
Last logged: 5:13:41 PM

Error communicating with FordPass for 3XXXXXXXXXXXXXXX1
Error communicating with FordPass for 1XXXXXXXXXXXXXXX7
```

```
This error originated from a custom integration.

Logger: custom_components.fordpass
Source: custom_components/fordpass/__init__.py:188
Integration: FordPass (documentation, issues)
First occurred: 2:57:00 PM (225 occurrences)
Last logged: 5:13:41 PM

local variable 'result' referenced before assignment
```

**Comment by circasurvivor55** [comment]
Dealing with the same error:

`This error originated from a custom integration.

Logger: custom_components.fordpass
Source: custom_components/fordpass/__init__.py:189
Integration: FordPass (documentation, issues)
First occurred: 5:33:44 PM (45 occurrences)
Last logged: 6:28:34 PM

Error communicating with FordPass for 1XXXXXXXXXXXXXXX7`

**Comment by itchannel** [comment]
Unfortunately it appears Ford has updated the API and in doing so have changed how the API authenticates. I'll have to look into it and rewrite a number of bits but it won't be a quick fix for now.

**Comment by 400HPMustang** [comment]
Unfortunate but understandable. I’m going to disable the integration for now and keep an eye out for an update from you.

**Comment by boostin4hp** [comment]
> Unfortunately it appears Ford has updated the API and in doing so have changed how the API authenticates. I'll have to look into it and rewrite a number of bits but it won't be a quick fix for now.

Would greatly appreciate it when you get a chance I love this integration!

**Comment by BluThunder2k** [comment]
Thanks for looking into it.     My vehicle as well stopped working with the same errors.

**Comment by JVTEAM** [comment]
Darn.  Thanks for the update.  Keep us posted.

**Comment by itchannel** [comment]
If anyone is willing to test I have a new beta version with fixed Auth flow, I'm currently running it on my HA to test for bugs but anyone else is welcome to have a look. [Beta](https://github.com/itchannel/fordpass-ha/tree/1.32-beta)

As the Auth flow has changed you may need to delete and resign into the integration or at the very least delete the token file first!
As I'm still testing I haven't had a play with all eventualities. but it works :)

![image](https://user-images.githubusercontent.com/5840973/174710488-5e4fc44a-f2ee-49c7-8665-0aaf0e91e396.png)

**Comment by tyldum** [comment]
Great work! Updated and iit works!

**Comment by serlinGi** [comment]
great, TOP!

**Comment by BluThunder2k** [comment]
Working here, just had to re-enable it (had disabled the integration), download the beta and we're back!    Thanks for you time!!

**Comment by eric-ewilson** [comment]
It's working for me too. Thanks for the quick fix!

**Comment by marbindrakon** [comment]
Just updated from 1.30 to 1.32Beta and restarted HA. It appears to be pulling data without any further actions since I didn't re-authenticate or delete any token files.

**Comment by CEri1975** [comment]
Great work...

**Comment by DevTodd** [comment]
The only issue that I see is that Tire Pressures are reporting as 9505 for all tires using LBS.  Otherwise I haven't found any other issues.  Thank you very much for getting the update out so quick.

**Comment by oleg-fiksel** [comment]
Same here. Works with https://github.com/itchannel/fordpass-ha/tree/1.32-beta
Thanks for fixing this! ❤️

**Comment by GerdRuetten** [comment]
I testing the 1.32-beta.
One VIN i can connect without failure.
Two other VIN's are not work.

**Comment by itchannel** [comment]
@GerdRuetten Can you tell me what kind of failure/error you are getting in the logs as I run multiple cars on mine and it's working fine.

**Comment by itchannel** [comment]
1.33 is now officially released and out of Beta (Had to skip 1.32 due to a git issue with Hacs)

**Comment by xpac1985** [comment]
Nice, thanks a ton for your work - that fixed it for me!
@itchannel @400HPMustang either of you might wanna close this issue ;)
